File: ios/Classes/SyncSubscriber.swift
import Flutter

internal struct SyncSubscriber: Equatable {
    private let channel: FlutterMethodChannel
    let masterId: String
    let subscriberId: String

    init(binaryMessenger: FlutterBinaryMessenger, masterId: String, subscriberId: String) {
        self.channel = FlutterMethodChannel(name: "dev.masel.synced_bloc.subscriber.\(masterId).\(subscriberId)", binaryMessenger: binaryMessenger)
        self.masterId = masterId
        self.subscriberId = subscriberId
    }

    func notifyMasterChange() {
        channel.invokeMethod("notifyMasterChange", arguments: nil)
    }

    static func == (lhs: SyncSubscriber, rhs: SyncSubscriber) -> Bool {
        return lhs.masterId == rhs.masterId && lhs.subscriberId == rhs.subscriberId
    }
}

File: android/src/main/kotlin/dev/masel/synced_bloc/SyncSubscriber.kt
package dev.masel.synced_bloc

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

internal class SyncSubscriber(
    binaryMessenger: BinaryMessenger,
    val masterId: String,
    val subscriberId: String
) {
    private val channel = MethodChannel(binaryMessenger, "dev.masel.synced_bloc.subscriber.$masterId.$subscriberId")

    fun notifyMasterChange() =
        channel.invokeMethod("notifyMasterChange", null)
}